go/processor/floria/state_contract.go

```diff
@@ -83,24 +83,24 @@ func isStateContract(address tosca.Address) bool {
 	return address == StateContractAddress()
 }
 
-// handleStateContract is a reworked version of the original function from the Opera client.
+// runStateContract is a reworked version of the original function from the Opera client.
 // It is used to handle epochs and allows to set balance, copy code, swap code, set storage, and increment nonce.
 // Source: https://github.com/Fantom-foundation/Sonic/blob/main/opera/contracts/evmwriter/evm_writer.go#L24
-func handleStateContract(
+func runStateContract(
 	state tosca.WorldState,
 	sender tosca.Address,
 	receiver tosca.Address,
 	input []byte,
 	gas tosca.Gas,
-) (tosca.CallResult, bool) {
+) tosca.CallResult {
 	if receiver != StateContractAddress() {
-		return tosca.CallResult{}, false
+		return tosca.CallResult{}
```

Comment:
Curiously, this used to return "false" if the recipient/receiver is of the wrong address. In such a case, in the old code, the state-contract invocation is ignored, since `isStatePrecompiled`, the second return value, is false, and the regular call processing is performed.

In the new version given here, the recipient is ignored, and the call fails. This looks like a change in semantic.